datafusion/src/row/reader.rs

```diff
@@ -44,6 +48,31 @@ pub fn read_as_batch(
     output.output().map_err(DataFusionError::ArrowError)
 }
 
+/// Read `data` of raw-bytes rows starting at `offsets` out to a record batch
+#[cfg(feature = "jit")]
+pub fn read_as_batch_jit(
```

Comment:
And this is the example usage pattern: generate code and compile once, and run repeatedly.